File: POO/Marketplace/Java/Cliente.java
class Cliente extends Entidade {

     private String nome;
     private String cpf;

     public Cliente(String nome, String cpf){
          this.nome = nome;
          this.cpf = cpf;
     }

     public Cliente(){
          System.out.println("Insira o nome do cliente");
          setNome(System.console().readLine());
          System.out.println("Insira o cpf do cliente");
          setCPF(System.console().readLine());
     }

     public String getNome(){
          return this.nome;
     }

     public void setNome(String nome){
          this.nome = nome;
     }

     public String getCPF(){
          return this.cpf;
     }

     public void setCPF(String cpf){
          this.cpf = cpf;
     }
}

File: CLP/TP2/cliente.rs
use Pessoa;
use pessoa::SetPessoa;
use pessoa::GetPessoa;

#[derive(Clone)]
pub struct Cliente {
    pub pes: Pessoa,
    pub rg: String,
    pub data_nascimento: String,
}

pub trait SetCliente {
    fn set_rg(&mut self, rg: String);
    fn set_data_nascimento(&mut self, data_nascimento: String);
}

pub trait GetCliente {
    fn get_rg(&self) -> String;
    fn get_data_nascimento(&self) -> String;
}

impl GetPessoa for Cliente {
    fn get_nome(&self) -> String{
        return self.pes.nome.clone();
    }

    fn get_endereco(&self) -> String{
        return self.pes.endereco.clone();
    }
}

impl GetCliente for Cliente {
    fn get_rg(&self) -> String{
        return self.rg.clone();
    }

    fn get_data_nascimento(&self) -> String{
        return self.data_nascimento.clone();
    }
}

impl SetPessoa for Cliente {
    fn set_nome(&mut self, nome: String){
        self.pes.nome = nome;
    }

    fn set_endereco(&mut self, endereco: String){
        self.pes.endereco = endereco;
    }
}

impl SetCliente for Cliente {
    fn set_rg(&mut self, rg: String){
        self.rg = rg;
    }

    fn set_data_nascimento(&mut self, data_nascimento: String){
        self.data_nascimento = data_nascimento;
    }
}
